hibernate-core/src/main/java/org/hibernate/internal/StatelessSessionImpl.java

```diff
@@ -491,7 +494,7 @@ public EntityPersister getEntityPersister(String entityName, Object object)
 			throws HibernateException {
 		checkOpen();
 		if ( entityName == null ) {
-			return getFactory().getMetamodel().entityPersister( guessEntityName( object ) );
```

Comment:
Hrm. This seems to not change behavior, but only because `guessEntityName()` has a different implementation on `StatelessSession`. I guess it's fine, but it just seems a little anomalous.

Reply:
I agree. In this PR I'm exclusively looking at the performance and am aiming for not making semantic changes, but there's something to look into better as a follow up.

Reply:
Yeah, so my concern was really that deleting that call to `guessEntityName()` might make it harder to notice the difference in future, but really it's no big deal, just a comment.